alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,269 @@
+# TargetDown
+
+## Meaning
+
+The alert means that Prometheus fails to scrape some targets.
+It fires when at least 10% of scrape targets in a Service are unreachable (`up`
+metric = 0) for at least 15 minutes.
+
+## Impact
+
+Metrics from the offline targets are not scraped by Prometheus. By consequence,
+we lose the visibility into the health and performance of the monitored application
+represented by the target.
+
+The alerts may lose accuracy or even become crippled if the missing metrics play
+a role in the triggering condition of the alert. Critical alerts may not fire
+when they should, potentially leading to unnoticed service disruption.
+
+Auto-scalers may not work properly if the missing metrics are critical to scaling
+conditions. This leads to potential degradation in user experience or waste of
+valuable resources.
+
+## Diagnosis & Mitigation
+
+The `TargetDown` alert indicates that one or more scrape targets are not responding
+to Prometheus scrape query.
+
+The alert has labels `namespace`, `service` and `job`. With these labels we can
+identify which Prometheus instance fails to scrape which target.
+
+In the Openshift web console, go to the tab **Observe** -> **Metrics**.  Check
+the metric `up` with these labels to find the down targets are in which pods.
+Here is an example query:
+```text
+up{namespace="openshift-monitoring", service="thanos-querier", job="thanos-querier"} == 0
```

Comment:
check the alert TargetDown details from "Observe -> Alerting" directly, would know which target is down

Reply:
> maybe we should just ask to run `up == 0` to get all the down targets (to avoid copy/paste this)

FYI, if set `up == 0` only, there are too many labels, see picture for `up == 1`
![Screenshot from 2023-09-19 13-30-09](https://github.com/juzhao/test/assets/22743651/8b7d5d4c-fa70-4861-9532-ae18f236f9c7)

Reply:
in some cases we may want to debug no-longer firing `TargetDown` alerts, I don't know if "Observe -> Alerting" keeps a history of them.

I was just looking for a way to get all the targets that are down without having the user deal with labels and to get a general view of targets being down as this may help establishing correlations.